File: solutions/code/Jury_Wolfris_C++/G.cpp
#include <iostream>

using namespace std;

int n, k, sum, maxx;

int main() {
    cin >> n >> k;
    for (int i = 0, a; i < n; ++i)
        cin >> a, sum += a, maxx = max(maxx, a);
    sum += k;
    if (sum % n == 0 && maxx <= sum / n)
        cout << "YES";
    else cout << "NO";
}

File: solutions/code/Jury_Wolfris_Pascal/g.pas
uses    math;
var n, k, i, sum, maxx: longint;
    a: array [1..100] of longint;
begin
    read(n, k);
    for i := 1 to n do
        read(a[i]);
    for i := 1 to n do
    begin
        inc(sum, a[i]);
        maxx := max(maxx, a[i])
    end;
    if ((sum + k) mod n = 0) and (maxx <= (sum + k) div n) then
        write('YES')
    else write('NO')
end.